packages/atomic/src/components/atomic-query-error/atomic-query-error.tsx

```diff
@@ -59,62 +50,87 @@ export class AtomicQueryError implements InitializableComponent {
   private renderShowMoreInfo() {
     if (!this.showMoreInfo) {
       return (
-        <button
+        <Button
           part="more-info-btn"
-          class="text-primary"
+          style="primary"
+          class="p-3 mt-8"
           onClick={() => (this.showMoreInfo = true)}
-        >
-          {this.strings.moreInfo()}
-        </button>
+          text={this.bindings.i18n.t('more-info')}
+        ></Button>
       );
     }
 
     return (
       <pre
         part="error-info"
-        class="text-left border border-neutral bg-neutral-light p-3 rounded my-4 whitespace-pre-wrap"
+        class="text-left border border-neutral bg-neutral-light p-3 rounded mt-8 whitespace-pre-wrap"
       >
         <code>{JSON.stringify(this.queryErrorState.error, null, 2)}</code>
       </pre>
     );
   }
 
+  private get org() {
+    return this.bindings.engine.state.configuration.organizationId;
+  }
+
+  private get errorType() {
+    return this.queryErrorState.error!.type;
+  }
+
+  private get icon() {
```

Comment:
I would refactor this to have one switch case for all "assets" (title, description, icon):
eg:
```
private getAssets() {
switch (this.errorType) {
      case disconnectedException:
        return {icon: NoConnection, title: this.bindings.i18n.t('disconnected'), description: this.bindings.i18n.t('check-your-connection'), link: undefined};
      case noEndpointsException:
        [...]
       // etc...
    }
}
```

It should be easier to mentally assemble what the end user will see for a given error. Also, a bit less repetition in the code.